TimoCloud-Universal/src/main/java/cloud/timo/TimoCloud/core/objects/Proxy.java

```diff
@@ -295,12 +296,12 @@ public boolean isConnected() {
         return connected;
     }
 
-    public DnsRecord getDnsRecord() {
-        return dnsRecord;
+    public ArrayList<DnsRecord> getDnsRecord() {
+        return dnsRecords;
     }
 
-    public void setDnsRecord(DnsRecord dnsRecord) {
-        this.dnsRecord = dnsRecord;
+    public void addDnsRecord(DnsRecord dnsRecord) {
```

Comment:
When are DNS records getting removed? A remove method should probably be called in CloudFlareManager:83

Reply:
ja das ist das einzige

Reply:
Ok but shouldn't the CloudFlareManager call a method `removeDnsRecord()` (which does not exist yet) on the proxy after removing the record via the CloudFlare API?

Reply:
im cloudflaremanger wird das alles removed, oder meinst du das die records auch aus der collection genommen werden?

Reply:
ja

Reply:
oh, das mache ich kurz rein

Reply:
for(Iterator<DnsRecord> iterator = proxy.getDnsRecords().iterator(); iterator.hasNext(); iterator.next()) {
                DnsRecord dnsRecord = iterator.next();
                deleteRecord(dnsRecord);
                iterator.remove();
            }

Reply:
so am besten oder wie soll ich das machen?

Reply:
Einfach mal eine Remove Methode in Proxy.java
```
public void removeDnsRecord(DnsRecord dnsRecord) {
    this.dnsRecords.remove(dnsRecord)
}
```

Mir fällt aber grad auf dass das ja sowieso nur aufgerufen wird, wenn der Proxy unregistered wird, dann brauchen wir nicht mehr aufräumen, wenn die Instanz dann ohnehin gleich gelöscht wird.

Reply:
dachte ich mir ja auch, also kann ich die weg lassen?

Reply:
ja